crates/oxc_linter/src/rules/eslint/id_length.rs

```diff
@@ -0,0 +1,629 @@
+
+                if self.is_exception(&ident_name) {
+                    return;
+                }
+
+                let graphemes_length = ident_name.graphemes(true).count();
+                let is_too_long = self.is_too_long(graphemes_length);
+                let is_too_short = self.is_too_short(graphemes_length);
+                if !is_too_long && !is_too_short {
+                    return;
+                }
+
+                if let Some(parent_node) = ctx.nodes().parent_node(node.id()) {
+                    match parent_node.kind() {
+                        AstKind::ExportSpecifier(_)
+                        | AstKind::ImportAttribute(_)
+                        | AstKind::ImportSpecifier(_)
+                        | AstKind::WithClause(_) => {
+                            return;
+                        }
+                        AstKind::ComputedMemberExpression(_)
+                        | AstKind::PrivateFieldExpression(_)
+                        | AstKind::StaticMemberExpression(_) => {
+                            let Some(parent_parent_node) =
+                                ctx.nodes().parent_node(parent_node.id())
+                            else {
+                                return;
+                            };
+                            let AstKind::SimpleAssignmentTarget(_) = parent_parent_node.kind()
+                            else {
+                                return;
+                            };
+
+                            if self.properties == PropertyKind::Never {
+                                return;
+                            }
+                        }
+                        AstKind::PropertyKey(property_key) => {
+                            if self.properties == PropertyKind::Never {
+                                return;
+                            }
+
+                            if let Some(parent_parent_node) =
+                                ctx.nodes().parent_node(parent_node.id())
+                            {
+                                match parent_parent_node.kind() {
+                                    AstKind::ObjectPattern(object_pattern) => {
+                                        // TODO: Is there a better way to do this check?
+                                        let binding_property_option = object_pattern
+                                            .properties
+                                            .iter()
+                                            .find(|x| x.key.content_eq(property_key));
+
+                                        if let Some(binding_property) = binding_property_option {
+                                            match &binding_property.value.kind {
+                                                BindingPatternKind::BindingIdentifier(_)
+                                                | BindingPatternKind::ObjectPattern(_) => {
+                                                    return;
```

Comment:
in line 248 there is an extra `if self.properties == PropertyKind::Never {`. 
Can both `AstKind::ObjectPattern` checks combined into one helper function?